Frarg parser refactor and github actions initialized.

AlbertoSoutullo opened the pull request:
Starting the refactor in a new branch called **FragRefactor**.

Firstly, I decided to refactor the parser, so I get familiarized with the flags and some behavior. Note that this tests are not made to ensure that the library argparser works ok, we assume that. We have to test that the flags that we will use are set correctly. That is why we mock some calls.

For example, in test_add_all_arguments, the only thing that add_all_argument does is call the inner functions to initialize the needed flags. **We do not want to call those methods**, because those methods have their own tests. So we just want to make sure that add_all_argument calls to it's inner functions. That is why we mock those calls.

Another example is test_check_highthroughput_in_args_true. We know that check_highthroughput_in_args_true needs a parameter that is an argument from argparser. We do not need to receive it as that, because the function will only change some parameters. So in this case, we mock the object, easily ensuring that the correct parameters are changed, independently of the object received.

Apart from this, I established a GitHub action workflow. It is tested in the latest Ubuntu, with only python 3.7. If more pythons are required to test, that can be easily changed in pythonapp.yml.

NostrumBioDiscovery commented:
Excellent (: Couple of things before approval:

1- Add nose to [frag_pele/conda_recipe/meta.yaml] (https://github.com/carlesperez94/frag_pele/blob/master/conda_recipe/meta.yaml) please.

2- Now that CI is beautifully implemented let's set that to work with python3.8/3.7/3.6/3.5 which are the one we support so far.

Dani

AlbertoSoutullo commented:
> Excellent (: Couple of things before approval:
> 
> 1- Add nose to frag_pele/conda_recipe/meta.yaml (https://github.com/carlesperez94/frag_pele/blob/master/conda_recipe/meta.yaml) please.
> 
> 2- Now that CI is beautifully implemented let's set that to work with python3.8/3.7/3.6/3.5 which are the one we support so far.
> 
> Dani

Now that you mention it, I forgot to remove nose from setup.py because nose is not used actually. I was testing some things, but finally I decided to use only unittest. In this last commit I removed Nose and added the rest of the Python versions.

It seems that Python 3.5 has some package conflicts. Will check it.
